### ISSUE TITLE
using salt.states.schedule.present always results in a change

### ISSUE THREAD

**Issue by sjorge** [saltstack/salt#30563]
using salt.states.schedule.present always results in a change of 'seconds: 0' if seconds parameter is not added!

```yaml
packages_daily_upgrades:  ## keep system up to date
  schedule.present:
    - function: pkg.upgrade
    - days: 1
    - splay: 300
```

```yaml
          ID: packages_daily_upgrades
    Function: schedule.present
      Result: True
     Comment: Modifying job packages_daily_upgrades in schedule
     Started: 20:30:01.332020
    Duration: 240.641 ms
     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -4,5 +4,4 @@
                   jid_include:True
                   maxrunning:1
                   name:packages_daily_upgrades
                  -seconds:0
                   splay:300
```

**Comment by jfindlay** [comment]
@sjorge, develop version?  Ping @garethgreenaway.

**Comment by sjorge** [comment]
Correct

**Comment by garethgreenaway** [comment]
Apologies for the delay on this one, able to duplicate the issue and working on a fix now.

**Comment by jfindlay** [comment]
Thanks, @garethgreenaway.

**Comment by sjorge** [comment]
Looks like it is fixed in my 2016.3 build from today. Marking this as closed.

**Comment by sjorge** [comment]
I stand corrected, it not yet fixed in 2016.3

**Comment by garethgreenaway** [comment]
The PR was made against develop, not 2016.3, it's likely it hasn't been merged in yet.  Would you mind testing against develop?

**Comment by sjorge** [comment]
I'm alternating between 2016.3 and develop. I probably messed up my builds earlier. I just grabbed my dev machine and it's fine in develop.

**Comment by sjorge** [comment]
@garethgreenaway spoke to soon:
```
----------
          ID: salt.minion::schedule_highstate
    Function: schedule.present
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "salt/state.py", line 1701, in call
                File "salt/loader.py", line 1607, in wrapper
                File "salt/states/schedule.py", line 184, in present
                File "salt/modules/schedule.py", line 137, in list_
              KeyError: 'seconds'
     Started: 23:40:14.246779
    Duration: 119.712 ms
     Changes:
----------
          ID: smartos.monitoring::schedule-fmadm
    Function: schedule.present
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "salt/state.py", line 1701, in call
                File "salt/loader.py", line 1607, in wrapper
                File "salt/states/schedule.py", line 184, in present
                File "salt/modules/schedule.py", line 137, in list_
              KeyError: 'seconds'
     Started: 23:40:14.367339
    Duration: 111.382 ms
     Changes:
----------
          ID: smartos.monitoring::schedule-zpool
    Function: schedule.present
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "salt/state.py", line 1701, in call
                File "salt/loader.py", line 1607, in wrapper
                File "salt/states/schedule.py", line 184, in present
                File "salt/modules/schedule.py", line 137, in list_
              KeyError: 'seconds'
     Started: 23:40:14.479700
    Duration: 145.468 ms
     Changes:
```

**Comment by sjorge** [comment]
@cachedout this one is another one that should go into 2016.3.1 I guess?

**Comment by rallytime** [comment]
@sjorge I have done a couple of merge-forwards in the last couple of weeks. The fix should be present on 2016.3 now. Can you confirm?

**Comment by sjorge** [comment]
@rallytime It's not available in 2016.3 yet. I just checked.

As mentioned by @garethgreenaway a few comments above the fix got committed to develop just after 2016.3 was branched. So merge-forwards would not have brought it into 2016.3.

**Comment by rallytime** [comment]
@sjorge Oh, my apologies! I thought @garethgreenaway's fix went into 2015.8. I misread.

**Comment by rallytime** [comment]
How about that @sjorge? ^^ :)

**Comment by sjorge** [comment]
@rallytime Yes! That would pull in the fix :)

**Comment by sjorge** [comment]
Fixed in 2016.3 and develop now, closing

**Comment by rallytime** [comment]
Thanks for following up! :)
